components/nimbus/src/enrollment.rs

```diff
@@ -2198,6 +2283,47 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_multi_feature_per_branch_conflict() -> Result<()> {
+        let mut test_experiments = get_test_experiments();
+        // conflicting_experiments conflict with themselves, but not with
+        // test_experiments
+        let mut conflicting_experiments = get_conflicting_multi_feature_experiments();
+        test_experiments.append(&mut conflicting_experiments);
+        let (nimbus_id, app_ctx, aru) = local_ctx();
+        let evolver = EnrollmentsEvolver::new(&nimbus_id, &aru, &app_ctx);
+        let (enrollments, events) =
+            evolver.evolve_enrollments(true, &[], &test_experiments, &[])?;
+
+        let enrolled_count = enrollments
+            .iter()
+            .filter(|&e| matches!(e.status, EnrollmentStatus::Enrolled { .. }))
+            .count();
+        // The two experiments in `conflicting_experiments` conflict with each
+        // other, so only one will be enrolled, plus the 2 from `test_experiments`
+        // is 3
+        assert_eq!(
+            enrolled_count, 3,
+            "There should be exactly 3 ExperimentEnrollments returned"
+        );
+
+        assert_eq!(
+            3,
+            events.len(),
+            "There should be exactly 2 enrollment_change_events"
+        );
+
+        let enrolled_events = events
+            .iter()
+            .filter(|&e| matches!(e.change, EnrollmentChangeEventType::Enrollment))
+            .count();
+        assert_eq!(
+            3, enrolled_events,
+            "exactly two events should have Enrolled event types"
+        );
+        Ok(())
+    }
+
```

Comment:
Aside: I hate how big our testing modules are, and how disorganized they are.

Also: do you think [`test_evolver_multi_feature_experiments`][1] needs adding to/refining/refactoring?

[1]: https://github.com/mozilla/application-services/blob/5774102b78c0a96b9acd80bf4c34701c85c725b3/components/nimbus/src/enrollment.rs#L2255